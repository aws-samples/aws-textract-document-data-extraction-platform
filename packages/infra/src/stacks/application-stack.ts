import { Stack, StackProps } from "aws-cdk-lib";
import { Construct } from "constructs";
import { addNagSupressionsToStack } from "../cfn-nag";
import { AuthStack } from "../components/auth/auth-stack";
import { PermissionsStack } from "../components/auth/permissions-stack";
import { SourceStack } from "../components/source/stack";
import { WebsiteStack } from "../components/website";

export class ApplicationStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    const authStack = new AuthStack(this, "AuthStack", {});
    const sourceStack = new SourceStack(this, "SourceStack", {
      userPool: authStack.userPool,
    });
    new WebsiteStack(this, "WebsiteStack", {
      authStack,
      sourceStack,
    });
    new PermissionsStack(this, "PermissionsStack", {
      userPool: authStack.userPool,
      identityPool: authStack.identityPool,
      sourceApi: sourceStack.sourceApi,
    });

    addNagSupressionsToStack(this);
  }
}
