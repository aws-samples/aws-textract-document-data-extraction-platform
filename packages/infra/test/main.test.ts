// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: MIT-0
import { App } from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";
import { ApplicationStack } from "../src/stacks/application-stack";

test("Snapshot", () => {
  const app = new App();
  const stack = new ApplicationStack(app, "test");

  const template = Template.fromStack(stack);
  expect(template.toJSON()).toMatchSnapshot();
});
