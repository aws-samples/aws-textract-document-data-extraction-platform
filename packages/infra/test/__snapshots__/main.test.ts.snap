// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
{
  "Metadata": {
    "cdk_nag": {
      "rules_to_suppress": [
        {
          "id": "AwsSolutions-S1",
          "reason": "overkill for this small sample",
        },
        {
          "id": "AwsSolutions-IAM4",
          "reason": "Managed policies are sufficient for a sample of this size",
        },
        {
          "id": "AwsSolutions-IAM5",
          "reason": "Some dynamic wildcard permissions are required for several service actions",
        },
        {
          "id": "AwsSolutions-DDB3",
          "reason": "Point in time recovery is not required",
        },
        {
          "id": "AwsSolutions-L1",
          "reason": "Lambda functions are using python 3.9",
        },
        {
          "id": "AwsSolutions-SF1",
          "reason": "Step functions not required to log ALL events. They are logged in lambda functions.",
        },
      ],
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "AuthStackNestedStackAuthStackNestedStackResourceC3CF3BFE": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "TemplateURL": {
          "Fn::Join": [
            "",
            [
              "https://s3.",
              {
                "Ref": "AWS::Region",
              },
              ".",
              {
                "Ref": "AWS::URLSuffix",
              },
              "/",
              {
                "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
              },
              "/6311e54ebed8ff3f626a4d07075b5bbded325f27bff1c2d40f3055161fb49194.json",
            ],
          ],
        },
      },
      "Type": "AWS::CloudFormation::Stack",
      "UpdateReplacePolicy": "Delete",
    },
    "PermissionsStackNestedStackPermissionsStackNestedStackResourceEC3A37F7": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "Parameters": {
          "referencetotestAuthStackNestedStackAuthStackNestedStackResourceFA57FC9BOutputstestAuthStackIdentityPool45395834Ref": {
            "Fn::GetAtt": [
              "AuthStackNestedStackAuthStackNestedStackResourceC3CF3BFE",
              "Outputs.testAuthStackIdentityPool45395834Ref",
            ],
          },
          "referencetotestSourceStackNestedStackSourceStackNestedStackResourceFC3341C5OutputstestSourceStackSourceApi9715424DRef": {
            "Fn::GetAtt": [
              "SourceStackNestedStackSourceStackNestedStackResourceEAAD4686",
              "Outputs.testSourceStackSourceApi9715424DRef",
            ],
          },
        },
        "TemplateURL": {
          "Fn::Join": [
            "",
            [
              "https://s3.",
              {
                "Ref": "AWS::Region",
              },
              ".",
              {
                "Ref": "AWS::URLSuffix",
              },
              "/",
              {
                "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
              },
              "/355c00a86cc1e6d1f11163595107d8fbb85fb6f859fbfe62b9845fba88fbaba0.json",
            ],
          ],
        },
      },
      "Type": "AWS::CloudFormation::Stack",
      "UpdateReplacePolicy": "Delete",
    },
    "SourceStackNestedStackSourceStackNestedStackResourceEAAD4686": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "Parameters": {
          "referencetotestAuthStackNestedStackAuthStackNestedStackResourceFA57FC9BOutputstestAuthStackUserPool60C3E5BFArn": {
            "Fn::GetAtt": [
              "AuthStackNestedStackAuthStackNestedStackResourceC3CF3BFE",
              "Outputs.testAuthStackUserPool60C3E5BFArn",
            ],
          },
        },
        "TemplateURL": {
          "Fn::Join": [
            "",
            [
              "https://s3.",
              {
                "Ref": "AWS::Region",
              },
              ".",
              {
                "Ref": "AWS::URLSuffix",
              },
              "/",
              {
                "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
              },
              "/b5bef8446a4895f554dd1b4534b2fef93795565c8d9157750cc032e4a960fcde.json",
            ],
          ],
        },
      },
      "Type": "AWS::CloudFormation::Stack",
      "UpdateReplacePolicy": "Delete",
    },
    "WebsiteStackNestedStackWebsiteStackNestedStackResourceC998E0CE": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "TemplateURL": {
          "Fn::Join": [
            "",
            [
              "https://s3.",
              {
                "Ref": "AWS::Region",
              },
              ".",
              {
                "Ref": "AWS::URLSuffix",
              },
              "/",
              {
                "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
              },
              "/2f35074acc44daf3a188f958f573785c4a365daad55af5382b26c3f31d90a5bc.json",
            ],
          ],
        },
      },
      "Type": "AWS::CloudFormation::Stack",
      "UpdateReplacePolicy": "Delete",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
